### PR TITLE
Add -Wunused-parameter

### PR DESCRIPTION
**Pull request by masterleinad**
Related to #2832 and #2796, superseding #2804.

### CODE REVIEW

**Comment by dalg24** [comment]
[CI passed on Jenkins](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/1360/pipeline/)

Not sure why it did not report back

**Comment by dalg24** [comment]
Retest this please

File: core/src/impl/Kokkos_ViewMapping.hpp [hunk 1]
```diff
@@ -3695,13 +3695,21 @@ KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType&) {
...
   return (size_t(i) < map.extent(R)) &&
          view_verify_operator_bounds<R + 1>(map, args...);
 }
+#else
+template <unsigned R, class MapType, class... Args>
+KOKKOS_INLINE_FUNCTION bool view_verify_operator_bounds(const MapType& /*map*/,
+                                                        Args... /*args*/) {
+  return true;
+}
+#endif
```
Comment (crtrott) [inline core/src/impl/Kokkos_ViewMapping.hpp]: I am not super happy with this change. Essentially it means for every data access the compiler has to inline/optimize away one more variadic function template call.

File: core/src/Kokkos_View.hpp [hunk 2]
```diff
@@ -821,21 +821,11 @@ class View : public ViewTraits<DataType, Properties...> {
     };
   };
```
Comment (crtrott) [inline core/src/Kokkos_View.hpp]: Goes with my other comment: this change increases the number of function inlines the compiler has to do for every data access. I don't really want that.

Comment (masterleinad) [reply core/src/Kokkos_View.hpp]: Let me see if the warning still appears and if I find a different solution.

Comment (masterleinad) [reply core/src/Kokkos_View.hpp]: I found a different solution.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by crtrott** [comment]
OK almost there. I think the KOKKOS_SINK should be KOKKOS_IMPL_SINK or what do other people think? Is this something which should be part of our public interface (in which case someone needs to write the documentation for it ...).

**Comment by masterleinad** [comment]
I don't think that this is something that needs to be exposed. In the end, we are only using it in this one place and only because we can't easily mark parameter packs as unused.
I'll just change to `KOKKOS_IMPL_SINK`.

**Comment by masterleinad** [comment]
Retest this please.

**Pull request merged**